int32 num_states
int32 num_actions
float32[] q_values